file-access.c

```diff
@@ -145,4 +145,34 @@ void save_final_scoreboard()
 
         return;
     }
-}
\ No newline at end of file
+}
+
+int count_question_files()
+{
+
+    int file_count = 0;
+    DIR *dirp;
+    struct dirent *entry;
+
+    dirp = opendir(QUESTION_PATH); 
+    if(dirp == NULL) {
+        perror("Question directory not found!");
+        exit(EXIT_FAILURE);
+    }
+    while ((entry = readdir(dirp)) != NULL){
+        if (entry->d_type == DT_REG){ 
+            file_count++;
+        }
+    }
+    closedir(dirp);
+    return file_count;
+}
+
+void verify_question_count()
```

Comment:
this shouldn't be concealed in this function. Conditional statement should be put implicitly in main.